second_task/src/Weapon.ts

```diff
@@ -1 +1,70 @@
-// your code goes here
\ No newline at end of file
+import { Item } from './Item'
+
+export abstract class Weapon extends Item {
+  baseDamage: number;
+  damageModifier: number;
+  durabilityModifier: number;
+  baseDurability: number;
+  weight: number;
+  effectiveDamage: number;
+  effectiveDurability: number;
+
+  constructor(
+    name:string,
+    baseDamage: number,
+    baseDurability: number,
+    value: number,
+    weight:number
+    ) {
+    super(name, value, weight)
+    this.name = name;
```

Comment:
@dianalemen, you don't need to assign variable in the derived class if you already pass them to the parent.